database/services/cost.service.js

```diff
@@ -1,30 +1,51 @@
-const costRepository = require('../repositories/cost.repository');
+const { checkPassword, WrongPasswordError } = require("./password.service")
+const costRepository = require('../repositories/cost.repository')
+const { addToBalance } = require('../repositories/balance.repository')
 
-const getCostService = (req, res) => {
+const getCostsService = (req, res) => {
   return costRepository
-  .getCosts()
-  .then((costs) => {
-    if (costs) {
-      return res.status(200).json(costs);
-    }
-    return res.status(404).json({ error: 'No costs found' });
-  })
-  .catch(err => res.status(500).json({ error: err.message }));
-};
+    .getCosts()
+    .then((costs) => {
+      if (costs) {
+        return res.status(200).json(costs)
+      }
+      return res.status(404).json({ error: 'No costs found' })
+    })
+    .catch(err => res.status(500).json({ error: err.message }))
+}
 
 const createCostService = (req, res) => {
-  const { date, amount } = req.body;
-  return costRepository
+
+  checkPassword(req.body, res)
+    .then((_) => createCost(req, res))
+    .catch(error => {
+      if (error instanceof WrongPasswordError) {
+        console.error("Creating the cost was not possible, due to: " + error.message)
+        return res.status(403).json({ error: 'Something went wrong, please try again later.' })
+
+      } else {
+        throw error
+      }
+    })
+}
+
+const createCost = (req, res) => {
+
+  costRepository
     .createCost(req.body)
     .then((insertedCost) => {
+
       if (insertedCost) {
-        return res.status(200).json(insertedCost);
+        addToBalance(-req.body.amount)
+          .catch(e => { throw e })
+
+        return res.status(201)
```

Comment:
Nekem _response timed out_ hibát ad, mikor be akarok vinni egy új cost-ot. Miközben itt az addToBalancenál jól fut le a függvény, ha beleírok egy then-t, akkor belemegy és helyesen módosítja is az adatbázisban az egyenleget.

Reply:
Ezt majd le kellene tesztelned. A review reszekent legyszi szedd ki az en branchemet es azon probald ki a funkciot.